### DISCUSSION TITLE
AttributeError: 'Symbol' object has no attribute 'DiracDelta'

### DISCUSSION THREAD

**Discussion by NwaObed** [zwicker-group/py-pde#253] [Q&A]
I am trying to use $\delta(x)$ as initial condition but i am getting the following error: 

ValueError                                Traceback (most recent call last)
ValueError: Error from parse_expr with transformed code: "Symbol ('sympy' ).DiracDelta (Symbol ('x' ))"

The above exception was the direct cause of the following exception: AttributeError: 'Symbol' object has no attribute 'DiracDelta'

Below is my code. What am I doing wrong? Even if i use 'DiracDelta(x)' instead of 'sympy.DiracDelta' since expressions are passed using the sympy package, I still have the same error. I need help please.

grid = pde.CartesianGrid([[-1, 1]], 64)  # generate grid
field = pde.ScalarField.from_expression(grid,'sympy.DiracDelta(x)')  # generate initial condition

eq = pde.PDE({"p": "d_dx(p * (a+(b*x))) + (0.05 * laplace(p))"},
             consts={'a':0.02,
                     'b':0.13},
             bc={"value": 0})
res = eq.solve(field, 15, dt=0.01,)

**Comment by david-zwicker** [comment]
The Dirac delta is not supported by `py-pde` – it's not a real function anyway. You can substitute with a very narrow peaked distribution, e.g., a Gaussian, or you construct the field like so
```python
grid = pde.CartesianGrid([[-1, 1]], 64)
field = pde.ScalarField(grid)
field.insert([0], 1)
```
where the last line inserts a certain integral value (in this case `1`) at the specified position (here `[0]`). This method uses linear interpolation, so it would also work when the position is not exactly at a support point.

**Reply by NwaObed** [reply]
Oh great! Thanks
